### ISSUE TITLE
Node.js 12 infrastructure checklist

### ISSUE THREAD

**Issue by rvagg** [nodejs/build#1768]
## Release

 - [x] **aix-ppc64**: AIX 7.1 TL05 on PPC64BE with GCC 6
 - [ ] **darwin-x64**: macOS 10.13, Xcode 10 https://github.com/nodejs/build/issues/1732
   - [x] `-mmacosx-version-min=10.10` / `'MACOSX_DEPLOYMENT_TARGET': '10.10'` in common.gypi - pending finalisation in https://github.com/nodejs/node/pull/26714
 - [x] **linux-arm64**: CentOS 7 with devtoolset-6 / GCC 6 - status quo is OK?
 - [x] **linux-armv7l**: New custom GCC 6 toolchain from https://github.com/rvagg/rpi-newer-crosstools (needs newer version of 6) https://github.com/nodejs/build/issues/1731 https://github.com/nodejs/build/pull/1776
 - [x] **linux-ppc64le**: Ubuntu 14.04 with GCC 6 https://github.com/nodejs/build/pull/1723
 - [x] **linux-s390x**: RHEL 7 with GCC 6 https://github.com/nodejs/build/pull/1735 https://github.com/nodejs/build/pull/1754
 - [x] **linux-x64**: CentOS 7 with devtoolset-6 / GCC 6 https://github.com/nodejs/build/pull/1766 https://github.com/nodejs/build/pull/1765 https://github.com/nodejs/build/pull/1763
 - [x] **sunos-x64**: SmartOS 18 with GCC 6 https://github.com/nodejs/build/issues/1733 https://github.com/nodejs/build/pull/1769
 - [x] **win-x64** and **win-x86**: Windows 2012 R2 (x64) with Visual Studio 2017 - status quo is OK?
 - [x] Switch to Linux for creating headers and source tarballs (only for 12+) https://github.com/nodejs/build/issues/1730

## Test

 - [x] AIX: tests only GCC 6 for Node 12+
 - [ ] macOS: 10.13 with Xcode 10 in CI mix for all, only 10.12 for Node 12+ https://github.com/nodejs/build/issues/1732
 - [x] ARM64: status quo is OK?
 - [x] ARMv7: GCC 4.8, 4.9 and 6 in CI mix, only GCC 6 for 12+ https://github.com/nodejs/build/issues/1731 https://github.com/nodejs/build/pull/1776
   - [x] scaleway armv7 machines running gcc 6
 - [x] ARMv6: removed from CI for Node 12+ only
 - [x] Linux-PPC: tests only GCC 6 for Node 12+
 - [x] Linux-s390x: tests only GCC 6 for Node 12+ https://github.com/nodejs/build/pull/1735 https://github.com/nodejs/build/pull/1754
 - [ ] Linux-x64: devtoolset-6 for 12+, continues testing devtoolset-2 for older versions
 - [x] SmartOS: 18 in CI for all versions (at least 12+) https://github.com/nodejs/build/issues/1733 https://github.com/nodejs/build/pull/1769
 - [x] Windows: status quo is OK?

## Miscellaneous

 - [ ] Land BUILDING.md changes @ https://github.com/nodejs/node/pull/26714
   - [ ] Python 3 & 2 support questions resolved as per https://github.com/nodejs/node/issues/25789
 - [ ] unofficial-builds to support ARMv6 community: not strictly necessary for 12.0.0 but communications need to be in place at least

**Comment by rvagg** [comment]
@nodejs/build I need some help in filling out the OP here with links and info and also getting these items ticked off the list. There's a bunch that are ready for ticking but need confirmation from the relevant expert(s). There's also a bunch that haven't even been started or are only partially done. If you want to put up your hand for something but don't have access, speak up and we'll see if it can be arranged.

**Comment by mhdawson** [comment]
I've checked the ones for IBM platforms.  We are still working to see if we can get centos in place on PPC, however, the current plan is to switch over after 12.x is cut.  GCC6 is installed on the existing PPC machines and it is being used to build/test for 12.x and higher. AIX was already using GCC6 and we have installed and are now using the devtoolset 6 on s390 for 12.x and higher.

**Comment by rvagg** [comment]
I'm pretty sure we have x64 sorted out now. I have smartos and armv7 in progress, have updated OP to link to the PRs for them.

**Comment by rvagg** [comment]
@joaocgreis would you mind ticking off the windows items if you think they're good as they are?

**Comment by joaocgreis** [comment]
Done, AFAICT Windows is ready.

**Comment by rvagg** [comment]
Getting very close, I have a smartos18 ansible PR that needs landing, one for the scaleway machines to only run Debian stretch, I (or someone else) needs to cast another eye over linux-x64 to confirm that only gcc-6 and devtoolset-6 are only being invoked for Node 12 (this isn't super critical but it'd be nice to have a smooth path when we start getting new libstdc++ features in and gcc-4.9 breaks). BUILDING.md is almost done. There's a PR for `-mmacosx-version-min=10.10`.

The two most critical pieces outstanding are: macOS 10.13 for test and release, and Python 2/3 stuff. 

Unofficial-builds comes up after all that as non-critical, and I'll be moving my focus back over to that once all of my other active stuff is sorted out.

**Comment by rvagg** [comment]
Time to draw a line under this stuff. As per https://github.com/nodejs/node/pull/26714#issuecomment-484726019 we've failed on Python 3 and macOS 10.13 + Xcode 10. The rest seems to be mostly in order. There is a need to check over all our Linux x64 test machines to make sure that Node 12+ is being compiled with GCC 6+ but I don't consider this critical for release, just something that should be done before we discover it broken with a V8 upgrade.
